fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -374,6 +380,28 @@ public void alterTable(ConnectContext context, AlterTableStmt stmt) throws StarR
         }
     }
 
+    @Override
+    public void truncateTable(TruncateTableStmt truncateTableStmt, ConnectContext context) {
+        String dbName = truncateTableStmt.getDbName();
+        String tableName = truncateTableStmt.getTblName();
+        org.apache.iceberg.Table table = icebergCatalog.getTable(context, dbName, tableName);
+
+        if (table == null) {
+            throw new StarRocksConnectorException(
+                    "Failed to load iceberg table: " + truncateTableStmt.getTblRef().toString());
+        }
+
+        DeleteFiles deleteFiles = table.newDelete().deleteFromRowFilter(Expressions.alwaysTrue());
+        deleteFiles.set(STARROCKS_USER, context.getCurrentUserIdentity() != null ?
```

Comment:
How about adding some engine infomations? Some execution engines, such as Spark or Flink, fill in engine-related information as snapshot tags, like `engine-name` and `engine-version`. For more details, please refer to the [Iceberg specification](https://iceberg.apache.org/spec/#other-fields)